### PR TITLE
Bug 1131261 - Make the speed dial operation asynchronous and delay showing the waiting overlay

### PR DESCRIPTION
**Pull request by gabrielesvelto**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[gabrielesvelto Gabriele Svelto (gabrielesvelto)](https://github.com/gabrielesvelto) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=07a14c0aed01)

File: apps/communications/dialer/locales/shared.en-US.properties [hunk 1]
```diff
@@ -1 +1,3 @@
 voiceMail=Voicemail
+noContactsWereFound=No contacts were found
+noContactsFoundDialogOk=ok
```
Comment (JohanLorenzo) [inline apps/communications/dialer/locales/shared.en-US.properties]: Capital letters would be used if we want to be consistent with the other "OK" buttons.

File: apps/communications/dialer/test/unit/keypad_test.js [hunk 2]
```diff
@@ -801,6 +801,8 @@ suite('dialer/keypad', function() {
         test('0# is ignored', function(done) {
           subject._getSpeedDialNumber('0').then(function(number) {
             assert.ok(false, 'the promise should be rejected');
+          }, function(error) {
+            assert.equal(error, 'noContactsWereFound');
```
Comment (tamarahills) [inline apps/communications/dialer/test/unit/keypad_test.js]: Could we add a spy to this test maybe to check that the ConfirmDialog gets called with noContactsFoundDialogOk?

**Comment by try-server-hook** [comment]
[gabrielesvelto Gabriele Svelto (gabrielesvelto)](https://github.com/gabrielesvelto) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=1a27ecd0e7c2)

**Comment by try-server-hook** [comment]
[gabrielesvelto Gabriele Svelto (gabrielesvelto)](https://github.com/gabrielesvelto) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=86663fd9d24d)

**Pull request merged**